File: API/Models/Item.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Models
{
    public class Item
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string id { get; set; }
        [BsonElement("type")]
        public string type { get; set; }
        [JsonProperty(PropertyName = "part_number")]
        [BsonElement("part_number")]
        public string partNumber { get; set; }
        [JsonProperty(PropertyName = "qty_on_hand")]
        [BsonElement("qty_on_hand")]
        public int? qtyOnHand { get; set; }
        [BsonElement("location")]
        public string location { get; set; }
        [BsonElement("height")]
        public double? height { get; set; }
        [BsonElement("width")]
        public double? width { get; set; }
        [BsonElement("length")]
        public double? length { get; set; }
        [BsonElement("uom")]
        public string uom { get; set; }
        [BsonElement("manufacturer")]
        public string manufacturer { get; set; }
        [JsonProperty(PropertyName = "manufacturer_number")]
        [BsonElement("manufacturer_number")]
        public string manufacturerNumber { get; set; }
        [JsonProperty(PropertyName = "used_for")]
        [BsonElement("used_for")]
        public string usedFor { get; set; }
        [BsonElement("horsepower")]
        public double? horsepower { get; set; }
        [BsonElement("amperage")]
        public double? amperage { get; set; }
        [BsonElement("voltage")]
        public double? voltage { get; set; }
        [BsonElement("rpm")]
        public int? rpm { get; set; }

    }
}

File: Client/src/app/Models/Item.ts
export class Item {
    _id: string;
    part_number: string;
    type: string;
    qty_on_hand: number;
    location: string;
    height: number;
    width: number;
    length: number;
    UOM: string;
    manufacturer: string;
    manufacturer_number: string;
    used_for: string;
    horsepower: number;
    amperage: number;
    voltage: number;
    RPM: number;
    
    constructor() {
        this._id = "";
        this.part_number = "";
        this.type = "";
        this.qty_on_hand = 0;
        this.location = "";
        this.height = 0;
        this.width = 0;
        this.length = 0;
        this.UOM = "";
        this.manufacturer = "";
        this.manufacturer_number = "";
        this.used_for = "";
        this.horsepower = 0;
        this.amperage = 0;
        this.voltage = 0;
        this.RPM = 0;
    }
}